tests/dragonfly/snapshot_test.py

```diff
@@ -592,16 +592,20 @@ async def test_big_value_serialization_memory_limit(df_factory, cont_type):
         f"debug populate 1 prefix {element_size} TYPE {cont_type} RAND ELEMENTS {elements}"
     )
 
+    await asyncio.sleep(1)
+
     info = await client.info("ALL")
     # rss double's because of DEBUG POPULATE
-    assert info["used_memory_peak_rss"] > (one_gb * 2)
+    peak_rss_before_save = info["used_memory_peak_rss"]
+    assert peak_rss_before_save > (one_gb * 2)
     # if we execute SAVE below without big value serialization we trigger the assertion below.
     # note the peak would reach (one_gb * 3) without it.
     await client.execute_command("SAVE")
     info = await client.info("ALL")
 
-    upper_limit = 2_250_000_000  # 2.25 GB
-    assert info["used_memory_peak_rss"] < upper_limit
+    # verify that the big value serialization mechanism is working
+    # after executing the SAVE command, there should be no spike in RSS memory.
```

Comment:
The assertion fail is
>       assert info["used_memory_peak_rss"] < upper_limit
E       assert 4905562112 < 2250000000

How can it be that used_memory_peak_rss is almost 5G? why do we fail only for steams type?
when you remove in your PR the upper_limit = 2.25G we dont check the rss against any constant so we dont know if our rss usage is reasonable
Also see the command in line 601
    # if we execute SAVE below without big value serialization we trigger the assertion below.
    # note the peak would reach (one_gb * 3) without it.

Reply:
`DEBUG POPULATE` uses more memory for streams, so we exceed the `upper_limit` before we even run `SAVE`

Reply:
Now I set the upper_limit based on the initial RSS memory peak before the SAVE command

Reply:
Why does debug populate uses more memory for streams?
If you set the upper_limit based on the initial RSS memory peak you can not really test that we do not spike in rss due to save, which is the purpose of this test.
In this test we want to check that rss peak is limited and low when we use big value serialization

Reply:
> Why does debug populate uses more memory for streams?
When I'm running this test locally: `Memory peak before save: 3096735744`.
So, it already cross the upper_limit.
DEBUG POPULATE uses significantly more memory for streams. I think we need to fix this, but it’s a separate issue that should be addressed in another PR.

Reply:
I will create a pr to fix debug populate cosuming more rss.
Right now I suggets to skip the stream type test case because the changes you introduce makes this test miss the test goal